include/nighthawk/common/termination_predicate.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/common/pure.h"
+
+namespace Nighthawk {
+
+class TerminationPredicate;
+
+using TerminationPredicatePtr = std::unique_ptr<TerminationPredicate>;
+
+/**
+ * Determines if and how a test is terminated.
```

Comment:
Do we have consistent terminology for what constitutes a "test"? https://github.com/envoyproxy/nighthawk/blob/master/api/client/service.proto talks about executions. Would be good to have a standard lexicon.

Reply:
Right, good point. Shall I rename `test` here to execution, and create a TODO to put up a `TERMINOLOGY.md` in the root of the repo?

Reply:
Sounds good to me.